src/modeField.js

```diff
@@ -0,0 +1,524 @@
+var ModelBase, ModelField, ModelOption, Mustache, globals, jiff,
```

Comment:
As discussed with @balihoo-wwilkins , it's good to add js files temporarily to PR so we don't lose work in progress.
This file should be removed before merging PR.